File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
class Solution {
    ArrayList<Integer> list = new ArrayList<>();

    public Object[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};

        int complete[] = new int[progresses.length];

        for(int i=0; i<progresses.length; i++){
            int nClear = 100 - progresses[i];
            complete[i] = nClear % speeds[i] == 0 ? nClear / speeds[i] : nClear / speeds[i] + 1;
        }

        int cnt = 1;
        int cur = complete[0];
        for(int i=1; i<complete.length; i++){
            if(cur >= complete[i]){
                cnt ++;
            }else{
                list.add(cnt);
                cur = complete[i];
                cnt = 1;
            }
        }
        list.add(cnt);

        return list.toArray();
    }
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.kt
class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        var answer = mutableListOf<Int>()

        val complete = mutableListOf<Int>()

        for(i in 0 until progresses.size){
            val nonClear = 100 - progresses[i]
            val pr = if(nonClear % speeds[i] == 0) nonClear/speeds[i] else nonClear/speeds[i] + 1
            complete.add(pr)
        }

        var cnt = 1
        var cur = complete.get(0)
        for(i in 1 until complete.size){
            if(complete.get(i) <= cur)
                cnt++
            else{
                answer.add(cnt)
                cnt = 1
                cur = complete.get(i)
            }
        }
        answer.add(cnt)

        return answer.toIntArray()
    }
}